### PR TITLE
feat: Add Modal Dialog

### PR DESCRIPTION
**Pull request by UKHOGreg**
<details>
  <summary>📦 Published PR as canary version: <code>0.14.0--canary.176.2e04e91.0</code></summary>
  <br />

  :sparkles: Test out this PR locally via:
  
  ```bash
  npm install @ukho/admiralty-angular@0.14.0--canary.176.2e04e91.0
  npm install @ukho/admiralty-core@0.14.0--canary.176.2e04e91.0
  # or 
  yarn add @ukho/admiralty-angular@0.14.0--canary.176.2e04e91.0
  yarn add @ukho/admiralty-core@0.14.0--canary.176.2e04e91.0
  ```
</details>

### CODE REVIEW

File: packages/website/src/app/components/modal-dialog/modal-dialog.mdx [hunk 1]
```diff
@@ -0,0 +1,62 @@
+# Modal Dialog
+
+Use modal dialogs to ?
+
+## Contents
+
+## Basic
+
+A.
```
Comment (ecstaticrainbow) [inline packages/website/src/app/components/modal-dialog/modal-dialog.mdx:9]: We need some actual words in here

File: packages/website/src/app/components/modal-dialog/modal-dialog.mdx [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+import Basic from "@/usage/modal-dialog/basic/index.mdx";
+
+<Basic />
+
+## Hidden
+
+A.
```
Comment (ecstaticrainbow) [inline packages/website/src/app/components/modal-dialog/modal-dialog.mdx:17]: Same here

**Review by ecstaticrainbow — CHANGES_REQUESTED** [review]

File: packages/website/src/app/components/modal-dialog/modal-dialog.mdx (hunk 1 above)
Comment (UKHOGreg) [reply packages/website/src/app/components/modal-dialog/modal-dialog.mdx:9]: Done

File: packages/website/src/app/components/modal-dialog/modal-dialog.mdx (hunk 2 above)
Comment (UKHOGreg) [reply packages/website/src/app/components/modal-dialog/modal-dialog.mdx:17]: Done

**Review by ecstaticrainbow — DISMISSED** [review]
Just one thing regarding the slot JSDoc

File: packages/core/src/components/modal-dialog/modal-dialog.tsx [hunk 3]
```diff
@@ -0,0 +1,42 @@
+import { Component, Prop, h } from '@stencil/core';
+
+@Component({
```
[the remark this answers is not in the document]
Comment (UKHOGreg) [reply packages/core/src/components/modal-dialog/modal-dialog.tsx:7]: Done!

**Review by ecstaticrainbow — APPROVED** [review]
LGTM

**Pull request merged**